File: activity-module/src/main/java/org/example/activitymodule/activity/domain/ActivityType.java
package org.example.activitymodule.activity.domain;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ActivityType {
    PHONE_CALL(1, "Phone call"),
    EMAIL(2, "Email"),
    MEETING(3, "Meeting"),
    TODO(4, "Todo"),
    NOT_AVAILABLE(0, "N/A");

    private final Integer code;
    private final String label;
}

File: ux/src/app/models/activity/activity-type.ts
import {faAt, faCheck, faMugHot, faPhone} from '@fortawesome/free-solid-svg-icons';

export const ActivityType = {
  PHONE_CALL: {code: 1, label: 'Phone call', icon: faPhone},
  EMAIL: {code: 2, label: 'Email', icon: faAt},
  MEETING: {code: 3, label: 'Meeting', icon: faMugHot},
  TODO: {code: 4, label: 'Todo', icon: faCheck}
}
